config/filter.d/mattermost.conf

```diff
@@ -0,0 +1,14 @@
+# Fail2Ban filter to match failed userlogin attempts as notified by mattermost's logfile
+#
+# Mattermost has to be configured to use timestamp at the start of the line. This is done in 'config/config.json'.
+# "LogSettings -> FileJson" needs to be set to 'false'
+# log lines should then look like this:
+# 2019-07-09T09:29:38.836+0200  error   mlog/log.go:172 Unable to find an existing account matching your credentials. This team may require an invite from the team owner to join.      {"path": "/api/v4/users/login", "request_id": "abc123", "ip_addr": "123.123.123.123", "user_id": "", "method": "POST", "err_where": "GetUserForLogin", "http_code": 400, "err_details": ""}
+
+[Definition]
+
+failregex = ip_addr.+"(?P<host>\d+.\d+.\d+.\d+)".+err_where.+(checkUserPassword|GetUserForLogin|ValidateToken).+
```

Comment:
This regex is extremely vulnerable:
- it is not anchored at all
- there are several catch-all's (`.+`)
- too few and not precise keywords/tokens should identify a failure, because can match everywhere in any message, so may be affected by injects, high cpu load or filter slowing down (up-to totally freezing) on some foreign input
- don't use `(?P<host>...)`, use `<ADDR>` (or `<HOST>` for older versions) instead of
- don't use capturing groups `(...)`, use `(?:...)` instead of
- etc.

So please use something like that:

```diff
- failregex = ip_addr.+"(?P<host>\d+.\d+.\d+.\d+)".+err_where.+(checkUserPassword|GetUserForLogin|ValidateToken).+
+ # used to bypass json key-value attribute(s):
+ _bypass_part = (?:"\w+"\s*:\s*(?:"[^"]*"|\S+)\s*,\s*)*
+ failregex = ^\s*error\s+[^\{]*\{\s*%(_bypass_part)s"ip_addr":\s*"<ADDR>"\s*,\s*%(_bypass_part)s"err_where"\s*:\s*"(?:checkUserPassword|GetUserForLogin|ValidateToken)"
```